### PR TITLE
Setup game mechanics

### PR DESCRIPTION
**Pull request by ThunderDev1**
Setup backend for managing game mechanics:
- start a round
- deals the cards
- hit the pile to end the round
- inform who won/lost

Multiple games can be played at the same time.

On the front end, there is a very basic display just to check everything is working.
This will be refactored in an other ticket.

### CODE REVIEW

File: server/Services/GameRound.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+using Microsoft.AspNetCore.SignalR;
+using Api.DTO;
+
+namespace Api.Services
+{
+  public class GameRound
+  {
+    private readonly int EggCards = 10;
+    private readonly int HenCards = 5;
+    private readonly int FoxCards = 5;
```
Comment (bellangerq) [inline server/Services/GameRound.cs]: Put the correct number of cards from the official rules (pinned message in Slack):
- 19 eggs
- 10 hens
- 10 [foxes](https://youtu.be/jofNR_WkoCE?t=39)

File: server/Services/GameRound.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+      Hub.Clients.Group($"game-{GameId}").SendAsync("receiveCard", card.ToString());
+    }
+
+    public void EndRound(string playerId)
+    {
+      GameTimer.Dispose();
+      // amount of eggs taking into account hens and foxes
+      int count = 0;
+      int activeChickens = 0;
```
Comment (bellangerq) [inline server/Services/GameRound.cs]: Are them chickens of hens? 🐔

File: server/Startup.cs [hunk 3]
```diff
@@ -32,10 +32,12 @@ public void ConfigureServices(IServiceCollection services)
           options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")
       ));
       services.AddTransient<IGameService, GameService>();
+      services.AddTransient<IGameManager, GameManager>();
       services.AddControllers();
 
       services.AddSignalR(options => options.EnableDetailedErrors = true);
 
+
```
Comment (bellangerq) [inline server/Startup.cs]: Remove extra line.

File: client/src/pages/Lobby/Lobby.tsx [hunk 4]
```diff
@@ -52,6 +67,23 @@ const Lobby: FC<LobbyProps> = (props: LobbyProps) => {
...
+    players: Array<Player>
+  ): string => {
+    if (!result.playerId) return `Personne n'a tapé sur la pile`
+    else {
+      let firstPart = ''
+      if (result.playerId === currentPlayerId) {
+        firstPart = 'Vous avez'
+      } else {
+        const player = players.find(player => player.id === result.playerId)
+        firstPart = `${(player && player.name) || 'Ta mère'} a`
```
Comment (bellangerq) [inline client/src/pages/Lobby/Lobby.tsx]: 😧

File: server/Services/GameRound.cs [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    private readonly int EggCards = 10;
+    private readonly int HenCards = 5;
+    private readonly int FoxCards = 5;
+
+    private Timer GameTimer;
+
+    private List<CardType> Deck { get; set; }
+    private List<CardType> Pile { get; set; }
+
+    private bool CorrectCount(int count) { return count == 5; }
```
Comment (bellangerq) [inline server/Services/GameRound.cs]: Shouldn't it be `>= 5`? If I hit while the count is 6, I am supposed to win.

File: client/src/pages/Lobby/Lobby.tsx [hunk 6]
```diff
@@ -93,6 +125,31 @@ const Lobby: FC<LobbyProps> = (props: LobbyProps) => {
             players={game.players}
             currentPlayerId={currentPlayerId}
           />
+          <div>
+            <h4>Cartes</h4>
```
Comment (bellangerq) [inline client/src/pages/Lobby/Lobby.tsx]: The previous heading is a `h2`. This one should either be a `h2` (same level) or a `h3` (lower level). Never skip heading levels!

And I think it should be a `h2` (even if we're going to redesign this section).

File: client/src/pages/Lobby/Lobby.tsx [hunk 7]
```diff
@@ -93,6 +125,31 @@ const Lobby: FC<LobbyProps> = (props: LobbyProps) => {
...
+            {cards.map((card, index) => (
+              <span key={index} style={{ paddingRight: '1rem' }}>
+                {card}
+              </span>
+            ))}
+            {result && (
+              <>
+                <h4>{buildResultText(result, game.players)}</h4>
+                <h4>{`Il y avait ${result.count} oeuf(s).`}</h4>
```
Comment (bellangerq) [inline client/src/pages/Lobby/Lobby.tsx]: Don't use headings for styling purposes!

File: server/Services/GameRound.cs (hunk 2 above)
Comment (ThunderDev1) [reply server/Services/GameRound.cs]: Hens 😄

File: server/Services/GameRound.cs (hunk 5 above)
Comment (ThunderDev1) [reply server/Services/GameRound.cs]: Ok, was not sure of that rule, I will update the wiki.

File: client/src/pages/Lobby/Lobby.tsx (hunk 7 above)
Comment (ThunderDev1) [reply client/src/pages/Lobby/Lobby.tsx]: I'll replace the h4 with a div + strong

File: server/Services/GameRound.cs (hunk 2 above)
Comment (bellangerq) [reply server/Services/GameRound.cs]: Then we should rename chickens into hens!

Comment (ThunderDev1) [reply server/Services/GameRound.cs]: Done

Comment (bellangerq) [reply server/Services/GameRound.cs]: You forgot `activeChickens`.

File: client/src/pages/Lobby/Lobby.tsx (hunk 7 above)
Comment (bellangerq) [reply client/src/pages/Lobby/Lobby.tsx]: A `<p>` is better for texts! Or even nothing, just the `<strong>`!

**Pull request merged**